### DISCUSSION TITLE
Does the build ignore your config-specific settings?

### DISCUSSION THREAD

**Discussion by martinwhitman** [dotnet/maui#26318] [Q&A]
I have a heck of a time switching between Debug and Release configurations or between iOS and Android. For instance, my prebuild command to switch app icons (to result in a similar appearance) between iOS and Android is often (but not always) ignored. Similarly, my config settings when switching between Debug and Release or between other configurations are often (but not always) ignored. My latest gripe is that my debug builds are still being optimized/linker-ed after switching back to debug from release. Can anyone give me any suggestions? Is this a Visual Studio issue? Am I struggling with something that is easy for the rest of you? Just looking for a little external perspective, thank you.

**Comment by TrueMoniker** [comment]
I can confirm the switching app icons does not always work. I find it will always replace the image file, but not necessarily before that portion of the app has been compiled. When I'm building for release, I always check to ensure my app icons are correct first.

One thing that seemed to reduce the amount of false-positives I had was telling the image swap to happen before anything else. I did this by creating an InitialTargets (CopyIcon) in my csproj file and creating additional targets that are told to run before my initial target (CopyIconByConfig)
```csproj
<Project Sdk="Microsoft.NET.Sdk" InitialTargets="CopyIcon">
        <!-- ...all the other app setup... -->
	<Target Name="CopyIcon"></Target>
	<Target Name="CopyIconByConfig" BeforeTargets="CopyIcon">
		<Copy SourceFiles="$(MSBuildProjectDirectory)\Resources\AppIcon\$(ConfigValue).svg" DestinationFiles="$(MSBuildProjectDirectory)\Resources\AppIcon\appicon.svg"></Copy>
	</Target>
</Project>

```
